src/main/java/com/easysplit/ess/transactions/infrastructure/persistence/TransactionsRepositoryImpl.java

```diff
@@ -160,4 +164,72 @@ public int getLastRevision(String debtGuid) {
 
         return revision;
     }
+
+    @Override
+    public TransactionEntity getTransaction(String transactionGuid){
```

Comment:
It seems to me that we are missing one method here, something like `getDebt(String transactionGuid)` to get the debt details for the transaction. This method would implement the query I pointed we are not using above